### PR TITLE
[SPARK-5712] [SQL] fix comment with semicolon at end

### PR DESCRIPTION
**Pull request by adrian-wang**
---- comment;

### CODE REVIEW

**Comment by adrian-wang** [comment]
retest this please.

**Comment by adrian-wang** [comment]
retest this please.

**Comment by adrian-wang** [comment]
cc @yhuai

**Comment by adrian-wang** [comment]
cc @rxin @marmbrus @liancheng

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveComparisonTest.scala [hunk 1]
```diff
@@ -241,7 +241,11 @@ abstract class HiveComparisonTest
       // Clear old output for this testcase.
       outputDirectories.map(new File(_, testCaseName)).filter(_.exists()).foreach(_.delete())
 
-      val allQueries = sql.split("(?<=[^\\\\]);").map(_.trim).filterNot(q => q == "").toSeq
+      val sqlWithoutComment =
+        sql.split("\n").filterNot(l => l.matches("--.*(?<=[^\\\\]);"))
+          .reduce((l1, l2) => l1 + "\n" + l2)
```
Comment (liancheng) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveComparisonTest.scala]: Seems this is enough?

```scala
val sqlWithoutComment = sql.split("\n").filterNot(_.startsWith("--")).mkString("\n")
```

**Comment by liancheng** [comment]
This LGTM except for a minor styling issue. Thanks!

**Comment by adrian-wang** [comment]
@liancheng , Thanks for your review! if you filter all the lines start with "--";
a query like
    -- abc
    select *;
would be trimmed as
    select *;
Then the hash code of the query would change, and that would lead to a lot of new golden files.

**Comment by adrian-wang** [comment]
Hi @liancheng , Can you take a look at this?

**Comment by adrian-wang** [comment]
ping @liancheng

**Comment by liancheng** [comment]
@adrian-wang Thanks for the explanation and sorry for the delay! I'm merging this to master.

**Pull request closed without merging**